jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -98,8 +98,41 @@ public static boolean getDefaultUseCaches()
         return __defaultUseCaches;
     }
 
+    public static Resource.Mount mountCollection(Collection<Resource> resources)
+    {
+        List<Resource.Mount> mounts = new ArrayList<>();
+        resources.forEach(res ->
+        {
+            Resource.Mount mount = Resource.mountIfNeeded(res);
+            if (mount != null)
+                mounts.add(mount);
+        });
+
+        try
+        {
+            return new ResourceCollection.Mount(resources, mounts);
+        }
+        catch (Throwable t)
+        {
+            // can't create ResourceCollection.Mount, unmount and rethrow.
+            mounts.forEach(IO::close);
+            throw t;
+        }
+    }
+
+    public static Resource.Mount mountCollectionURIs(Collection<URI> uris)
+    {
+        return mountCollection(uris.stream().map(Resource::newResource).toList());
```

Comment:
This won't work as if one URI does need mounting, trying to create a `Resource` from it will throw. This should call `mountIfNeeded(URI uri)` and return a combined mount.

Reply:
That's not how `Resource` currently works.

Reply:
Unit tests added, implementation greatly simplified.